### ISSUE TITLE
[FR] Documentation about Resolving OOM

### ISSUE THREAD

**Issue by tongyx361** [volcengine/verl#1014]
### Motivation

There are many issues related to OOM, e.g. #328 . We might need a clear guide about how to resolve OOM.

### Plan

A non-exclusive enumeration about related configurations:

1. Rollout：`gpu_memory_utilization`
2. Other Inference：
    1. Liger Kernel 
    2. `*_max_len_per_gpu` / `micro_batch_size_per_gpu`
3. Training:
    1. Liger Kernel
    2. Ulysses Sequence Parallelism
    3.  gradient checkpointing
    4. offload

### TODO

- [ ] Complete the list of related configurations
- [ ] Benchmark the effect & overhead of each configuration

**Comment by gameofdimension** [comment]
will also track cpu oom?

**Comment by tongyx361** [comment]
> will also track cpu oom?

Welcome to contribute if you need this feature!

**Comment by warmsnow-sh** [comment]
Very good advice! It is particularly suitable for people like us who work with RTX4090

**Comment by shirley-wu** [comment]
Thanks for putting together these suggestions! Any suggestions for resolving cpu OOM though? I'm always having cpu OOM for 7B model. Takes a few hundred GB cpu memory.

**Comment by gameofdimension** [comment]
> Thanks for putting together these suggestions! Any suggestions for resolving cpu OOM though? I'm always having cpu OOM for 7B model. Takes a few hundred GB cpu memory.

I fixed my CPU OOM issue by adopting [tcmalloc](https://www.hiascend.com/doc_center/source/zh/Pytorch/600/ptmoddevg/trainingmigrguide/performance_tuning_0067.html). Hope it works for you too!

**Comment by WindowsXp-Beta** [comment]
I tried enabling ulysses sequence parallelism in training but it resulted in a higher GPU memory usage. Any idea why this could happen?
w/o sequence parallelism
![Image](https://github.com/user-attachments/assets/37905da0-c5dc-486b-9338-2a0764f02f2c)
w/ sequence parallelism(sp = 8)
![Image](https://github.com/user-attachments/assets/717da17c-ece7-4860-92f7-95daffa8b556)

**Comment by Kirrito-k423** [comment]
定时清理检查结果：这个 issue 最近一次由当前运行账号以外账号产生的公开回复或公开引用是 `2025-12-29T15:24:20Z`。本轮检查日期为 2026-07-04，该时间已经超过 6 个月。检查来源包括 issue comments API `https://api.github.com/repos/verl-project/verl/issues/1014/comments?per_page=100` 和 timeline API `https://api.github.com/repos/verl-project/verl/issues/1014/timeline?per_page=100`。

按照本仓库的定时 issue 清理标准，我会先留下这条说明并关闭该 issue。若作者或社区成员仍能提供新的复现、证据或需求背景，请继续评论请求重开，或创建新的 issue 并引用这里的上下文。

---

Scheduled cleanup check: the latest public reply or public reference from an account other than the current automation account was `2025-12-29T15:24:20Z`. The check date for this run is 2026-07-04, so that activity is more than 6 months old. The checked public sources were the issue comments API `https://api.github.com/repos/verl-project/verl/issues/1014/comments?per_page=100` and the timeline API `https://api.github.com/repos/verl-project/verl/issues/1014/timeline?per_page=100`.

Following the scheduled issue cleanup policy, I am leaving this note first and then closing the issue. If the author or community can still provide a fresh reproduction, evidence, or requirement context, please comment to request reopening or open a new issue that links back to this one.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
